internal/api/event.go

```diff
@@ -0,0 +1,101 @@
+package api
```

Comment:
There are two main reasons:
1. The function is only used in internal packages. 
2. If I keep it in the original directory, importing `ArgoCDUpdateConfig` from the internal package will cause `import cycle not allowed`, as the `v1alpha1/` package is also imported in the internal package.

Reply:
I understand the import cycle you're trying to avoid, but I don't think this package is a good home for this. (idk if its old home was any better, tbh.)

"API" is overloaded. I wish _this_ package had been named `apiserver` instead, as this package is all API server endpoints, interceptors, etc. and this functionality you're moving to here is used much more broadly -- by the controller, by the webhooks, etc.

Is it possible this might be better off living in the existing `event` package (e.g. Live alongside things like `func NewRecorder()`)?

Reply:
@krancour I tried placing it in the existing `event` package, but realized it is better for k8s event type. Since I couldn’t find a better location, I’ve refactored some files into internal folders in https://github.com/akuity/kargo/pull/2808. PTAL if this makes sense. Thank you!